sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -51,18 +61,25 @@ def __init__(self, dataflow_client=None, job_result=None, job_graph=None):
     self._cached_metrics = None
     self._job_graph = job_graph
 
+  @staticmethod
+  def is_counter(metric_result):
+    return isinstance(metric_result.attempted, numbers.Number)
+
+  @staticmethod
+  def is_distribution(metric_result):
+    return isinstance(metric_result.attempted, DistributionResult)
+
   def _translate_step_name(self, internal_name):
     """Translate between internal step names (e.g. "s1") and user step names."""
     if not self._job_graph:
       raise ValueError('Could not translate the internal step name.')
 
     try:
-      [step] = [step
-                for step in self._job_graph.proto.steps
-                if step.name == internal_name]
-      [user_step_name] = [prop.value.string_value
-                          for prop in step.properties.additionalProperties
-                          if prop.key == 'user_name']
+      step = get_first(self._job_graph.proto.steps,
```

Comment:
There should only be one instance with `key = 'step'`. I added the function because I found myself 'querying' for an element in a list very often here. I will rename the function to `_get_match`, if that's fine by you.